### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by nrdxp**
The biggest change here is the cleansing of all outstanding clippy lint warnings. I had to add an exception for     `clippy::cast_possible_truncation` as the `lsp_types::Position` struct requires a u32, but we are currently building it from a usize.

I also ran a `cargo fmt` on the codebase, removed the implicit nix-direnv dependency on the envrc, added some rust development tools to the devshell and pulled rust from unstable to get a slightly newer version.

[One of the lints](https://github.com/nix-community/rnix-lsp/pull/52/files#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fcL151) corrected a scenario where a value's `Drop` destructor was never called, so there may be a miniscule chance that this was the root cause of #33.

### CODE REVIEW

File: .envrc
Comment (Ma27) [inline .envrc:11]: Can we somehow explicitly declare this dependency or does it make sense to at least document it in the README? I'm not sure if we want to maintain this on our own if there's an existing tool for that :)

File: src/main.rs
Comment (Ma27) [inline src/main.rs:43]: I don't think that this improves the readability. It seems as if that's a great way to get avoidable merge-conflicts especially if multiple folks edit this. So while adding `clippy` here by default is a great thing, I'd prefer to exclude this rule :)

**Review by Ma27 — COMMENTED** [review]
Thanks! Below are a few notes. I'll recompile my rnix-lsp with these changes and run it for a few days as well. As soon as we resolved the comments, I'd merge then.

Another idea: would you mind adding clippy (and perhaps even a style-check) to GH actions? That way, we can ensure that only changes with these tools being happy get merged into master (IIRC both tools have autofixers, so this shouldn't be an extra-effort for contributors). Also, we don't have to `cargo fmt` regularly which renders `git blame` unusable :) 

@fufexan did you have a chance to test this? As @nrdxp has pointed out, this fixes a another potential memory leak :)

**Comment by aaronjanse** [comment]
I like these changes. Thank you @nrdxp! I also agree with @Ma27's feedback :-)

**Comment by fufexan** [comment]
Just tried it out. Seems like a way to trigger it is to open a `let` binding but not close it, then add an `inherit`. Instant trigger.

**Comment by Ma27** [comment]
Wait.... I think... I can reproduce the memleak, realiably :tada: (and I kinda feel dumb now :D)

I halted both affected `rnix-lsp` processes with `gdb` on my laptop for now since I'm actually busy with studying right now, but I'll try to take a look later, thanks! :tada:

File: .envrc
Comment (nrdxp) [reply .envrc:11]: direnv does have a mechanism for for import nix-direnv if it doesn't exist. I just figured, since the functionality can be replicated in a few lines that it wasn't worth it.

File: src/main.rs
Comment (nrdxp) [reply src/main.rs:43]: I was considering this as well, but thought I'd just go along with clippy for the first round and see if there were any objections :slightly_smiling_face:

Comment (aaronjanse) [reply src/main.rs:43]: I'm personally in favor of using an explicit (for example) `lsp_types::CompletionItem` syntax over `use lsp_types::*` when there are many imports. But I don't want to nitpick too much, so I'm in favor of either just accepting this entire PR (then later refactoring slightly) or reverting this small chunk of the diff, whichever @Ma27 prefers

**Review by aaronjanse — APPROVED** [review]
Thank you @nrdxp!! I know that Clippy lints can be tedious to fix, so I appreciate you going through the effort to address them. I'm personally in favor of merging this PR as-is, but I'll defer to @Ma27 for deciding whether to include the long `use lsp_types::{ ... }` statement

**Comment by aaronjanse** [comment]
Also, I'm willing to fix the merge conflict while merging this PR, rather than making @nrdxp fix it (unless they want to)

**Comment by Ma27** [comment]
@nrdxp may I ask if you intend to finish this? Otherwise I'd take care of it when i'll get to it :)

**Comment by nrdxp** [comment]
sorry, I completely forgot about this. I wouldn't mind rebasing it and finishing it off, but if you have the time and the motivation, feel free to take over. If not, I'll schedule it for this weekend.

**Comment by Ma27** [comment]
> If not, I'll schedule it for this weekend

If you still have the capacity, feel free to finish :) 
I wouldn't have managed to take care of it last week (and most likely also not this weekend).

> . That said, it might be interesting to also bump the rust edition now that 2021 is stable, although it's a fairly minor change all things considered so it may not make much of a difference.

Oh, good point, thanks for the reminder!

Feel free to try it out :)

**Comment by nrdxp** [comment]
just fyi, I started on refactoring this on the weekend. Unfortunately didn't finish since I was rather busy with family stuff, but I'm hoping I should be able to get it done sometime in the next week.

**Pull request closed without merging**
